portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,21 +28,46 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  // adding hard-coded messages to arraylist
   ArrayList<String> sampleMessages = new ArrayList<>();
 
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {  
     String messages = convertToJsonString(sampleMessages);
     response.setContentType("text/html;");
     response.getWriter().println(messages);
   }
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the inpiut from the form.
+    String userComment = getUserComment(request);
+    if(userComment == null) {
+        response.setContentType("text/html");
+        response.getWriter().println("Please enter a valid comment");
+        return;
+    }
+    sampleMessages.add(userComment);
+    response.sendRedirect("/dinosaur.html");
+  }
+
+  private String getUserComment(HttpServletRequest request) {
+      // Get input from the form.
+      String userName = request.getParameter("user-name");
+      String userComment = request.getParameter("user-comment");
+
+      if(userName.length() <= 0) {
+          System.err.println("Please input name");
```

Comment:
shouldn't you return this as a error from the function?

Reply:
This function getUserComment() is some kind of a helper function whose return value will determine the behavior of the other function , doPost(). As you have mentioned in a previous suggestion, I think I should throw an exception in that function instead of just printing an error message. 
Since I need to make sure the value is not null, do I still need to return an error here?

Reply:
So I would say if the function is not going to be used by other handlers, then it's better to just make the code inline.  Since doPost is specifically handling user comments, it's okay to do the validation inside it directly and that way you can throw exception if validation fails.

If the function getUserComment is to be reused, then you should return an error or some form of exception (with details about the error) from within in and that can then be thrown as an exception in the doPost handler.